plugins/calendar/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/calendar/CalendarSpeech.java

```diff
@@ -62,32 +97,321 @@ public String getEvents(Map<String, EntityData> entities) {
+				withEndDate = !checkDate.isEqual(endDate);
+			} else {
+				// event ends on dayToCheck
+				withStartDate = true;
+			}
+		} else {
+			// event will start in future
+			outputCase = eventType(startDate, endDate, event, OutputCase.STARTINFUTURE);
+			// event will finish on an other day in the future
+			withEndDate = !startDate.isEqual(endDate);
+		}
+
+		return eventToString(startDateTime, endDateTime, event, withStartDate, withEndDate, withTime, outputCase);
+	}
+	
+	/**
+	 * This method formats the date and time of the events and adds them into the list
+	 *
+	 * @param startDate
+	 *            start time of the event
+	 * @param endDate
+	 *            end time of the event
+	 * @param event
+	 *            data of the event
+	 * @param withStartDate
+	 *            determines if event output is with or without the start date
+	 * @param withEndDate
+	 *            determines if event output is with or without the end date
+	 * @param withTime
+	 *            distinguishes if it is an event with or without a time stamp
+	 * @param outputCase
+	 *            distinguishes between different output cases
+	 * @return the string generated for the event
+	 */
+	public String eventToString(LocalDateTime startDate, LocalDateTime endDate, CalendarEvent event, boolean withStartDate,
+			boolean withEndDate, boolean withTime, OutputCase outputCase) {
+		String eventData = convertEventTitle(event.getSummary());
+		String eventStartDate = dateOutput(startDate, withStartDate, withTime);
+		String eventEndDate = dateOutput(endDate, withEndDate, withTime);
+		String eventStartTime = "";
+		String eventEndTime = "";
+		if (withTime) {
+			eventStartTime = " " + getCorrectTime(startDate);
+			eventEndTime = " " + getCorrectTime(endDate);
+		}
+		switch (outputCase) {
+		case STARTINPAST:
+			if (withStartDate || withTime) {
+				eventData += " since" + eventStartDate + eventStartTime;
+			}
+			if (withEndDate || withTime) {
+				eventData += " until" + eventEndDate + eventEndTime;
+			}
+			eventData += ". \n";
+			break;
+		case STARTINFUTURE:
+			eventData += " from" + eventStartDate + eventStartTime;
+			eventData += " until" + eventEndDate + eventEndTime + ". \n";
+			break;
+		case ALLDAYLONG:
+			if (withStartDate) {
+				eventStartDate = " on the " + getDate(startDate);
+			}
+			eventData += eventStartDate + " all day long. \n";
+			break;
+		case SINGLEDAY:
+			if (withStartDate) {
+				eventData += " on the " + getDate(startDate) + " at " + getCorrectTime(startDate) + " until "
+						+ getCorrectTime(endDate) + ". \n";
+			} else {
+				eventData += " from " + getCorrectTime(startDate) + " until " + getCorrectTime(endDate) + ". \n";
+			}
+			break;
+
+		default:
+			this.logger.error("Missing or wrong output case.");
+			break;
+		}
+
+		return eventData;
+	}
+
+	/**
+	 *
+	 * @param localDateTime
+	 *            input time as LocalDateTime
+	 * @return correct time in 24 hour format with am/pm
```

Comment:
Will this print 1pm or 13pm? 8am, 6pm etc is called 12 hour format afaik

Reply:
13 pm so that google says 1 pm but it is visualized in 24 hour format

Reply:
but I can change it to 12 hour format if it is preferred